### PR TITLE
Ieatkillerbees/ch80/child crud endpoints

### PR DESCRIPTION
**Pull request by ieatkillerbees**
This PR

- [x]  Adds a CreateChild endpoint
- [x]  Adds a ListChildren endpoint
- [x]  Adds a GetChild endpoint
- [x]  Enables global family tenancy filtering for routes under /families/

### CODE REVIEW

File: src/Actions/Child/CreateChild.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+use Choredo\Transformers\ChildTransformer;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Teapot\StatusCode\Http;
+use Zend\Diactoros\Response\JsonResponse;
+use const Choredo\REQUEST_RESOURCE;
+
+class CreateChild implements FractalAware, LoggerAwareInterface, EntityManagerAware
```
Comment (greydnls) [inline src/Actions/Child/CreateChild.php:20]: When a mommy controller and a daddy controller love each other very much......

File: src/Actions/Child/CreateChild.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            'New Child created',
+            ['id' => $child->getId()->toString(), 'familyId' => $child->getFamily()->getId()->toString()]
+        );
+
+        $item = $this->outputItem($child, new ChildTransformer(), 'children')->toArray();
+
+        return (new JsonResponse($item, Http::CREATED))
+            ->withHeader(
+                'location',
+                '/families/' . $child->getFamily()->getId()->toString() .
```
Comment (greydnls) [inline src/Actions/Child/CreateChild.php]: /pedantic/ These could be local variables to save you calling ->->->toString() twice

File: src/Entities/Child.php [hunk 3]
```diff
@@ -7,6 +7,7 @@
 use Assert\Assert;
 use Doctrine\ORM\Mapping as ORM;
 use Ramsey\Uuid\UuidInterface;
+use const Choredo\HEX_COLOR_REGEX;
```
Comment (greydnls) [inline src/Entities/Child.php:10]: A+++++ for this.

File: src/Actions/Register.php [hunk 4]
```diff
@@ -8,7 +8,7 @@
 use Choredo\EntityManagerAware;
 use Choredo\HasEntityManager;
 use Choredo\Output\CreatesFractalScope;
-use Choredo\Output\FractalAwareInterface;
+use Choredo\Output\FractalAware;
```
Comment (greydnls) [inline src/Actions/Register.php:11]: Probs shoulda done this in a separate PR. You know... if you cared about SRP.

File: src/Hydrators/ChildHydrator.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+              ->alnum()
+              ->length(6)
+              ->verifyNow()
+        ;
+    }
+
+    /**
+     * @return \Choredo\Entities\Family
+     */
+    public function getFamily(): Family
```
Comment (greydnls) [inline src/Hydrators/ChildHydrator.php]: I mean... if you're gonna make an interface, might as well make a trait. #justsayin

File: src/Hydrators/ChildHydrator.php [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+     * @var \Choredo\Entities\Family
+     */
+    private $family;
+
+    public function hydrate(JsonApiResource $resource)
+    {
+        $this->validateResource($resource);
+
+        return new Entities\Child(
+            $resource->getId() === JsonApiResource::TYPE_NEW ? Uuid::uuid4() : Uuid::fromString($resource->getId()),
```
Comment (greydnls) [inline src/Hydrators/ChildHydrator.php]: Might we worth throwing this in a static helper class or summat. 

`ResourceIdGenerator::generateId($resource)`

or a global function. whatevs.

File: src/Transformers/ChildTransformer.php [hunk 7]
```diff
@@ -13,23 +13,36 @@ class ChildTransformer extends TransformerAbstract
...
             'name'        => $child->getName(),
+            'color'       => $child->getColor(),
             'accessCode'  => $child->getAccessCode(),
             'avatarUri'   => $child->getAvatarUri(),
             'createdDate' => $child->getCreatedDate(),
             'updatedDate' => $child->getUpdateDate(),
         ];
+
+        if ($child->getCreatedDate()) {
+            $entity['createdDate'] = $child->getCreatedDate()->format(\DateTime::ATOM);
```
Comment (greydnls) [inline src/Transformers/ChildTransformer.php]: Why would these ever not be set?

File: src/Actions/Child/CreateChild.php (hunk 1 above)
Comment (ieatkillerbees) [reply src/Actions/Child/CreateChild.php:20]: :laughing:

**Review by greydnls — APPROVED** [review]
